node_modules/oae-content/lib/api.js

```diff
@@ -539,28 +539,77 @@ var _createContent = function(ctx, contentId, revisionId, resourceSubType, displ
             return callback(err);
         }
 
-        // The current user always becomes a manager
-        roles[ctx.user().id] = AuthzConstants.role.MANAGER;
-
-        // Create the resource
-        var createFn = _.partial(ContentDAO.Content.createContent, contentId, revisionId, ctx.user().id, resourceSubType, displayName, description, visibility, otherValues, revisionData);
-        ResourceActions.create(ctx, roles, createFn, function(err, content, revision, memberChangeInfo) {
+        // If the content is on a folder, add those members to the content created 
+        _getFolderMembers(ctx, folders, roles, resourceSubType, function(err, roles) {
```

Comment:
aha! here's the problem. These are the roles that are currently valid, but this is not what we want, is it? we want the permissions that have been defined by the user in the "shared with" form. Those are the ones we want.

A good strategy would be to pick the current ones and then filter them using the form. That way we would have the folder roles, e.g.

```
{ 'u:up:SkmX6OL9M': 'manager', 'u:uc:HyDW6O8qf': 'editor' }
```

and then compare it to what comes through the form. For instance, if one user has been removed from the input field inside the modal box, we take it out and it becomes 

```
{ 'u:up:SkmX6OL9M': 'manager' }
```

Reply:
The POST that is sent when one creates the document goes like this (in my local scenario):

```
resourceSubType:collabdoc
displayName:My private fifth document
description:
visibility:private
managers:u:up:SkmX6OL9M
managers:u:up:SkmX6OL9M
folders:f:up:rkqGJYUqG
```

So you see, that's the baseline we should be using: `{ 'u:up:SkmX6OL9M': 'manager' }`. The current folder permissions doesn't matter.

On another test, where I didn't remove any users from the "Shared with" field, this is what I got:

```
resourceSubType:collabdoc
displayName:My public document
description:
visibility:private
managers:u:up:SkmX6OL9M
managers:u:uc:HyDW6O8qf 👈 
managers:u:up:SkmX6OL9M
folders:f:up:rkqGJYUqG
```

...which confirms my approach: this is what we need `roles` to contain. 👍

Reply:
I finally removed all my code (no code, no mistake haha) because I get all members on 3akai-ux and there is no need to add code in Hilary but there is something that still bothers me... On a collabdoc, if a user is added on the "setpermissions" modal, they will be added as manager. 

-> So if there is 30 viewers on a folder, when a collabdoc is created inside the folder, this collabdoc will have 30 managers (+ the creator of the resource). 

For me, a viewer of a folder should be editor(or viewer) but not manager ... I don't know, that's just my point of view.

(Just for information, viewers and managers of folder will become viewers of a link created inside the folder.)